### PR TITLE
Remove usages of Prototype.js

### LINKED ISSUE #31: Remove usages of Prototype from Categorized View
See [this blog post](https://www.jenkins.io/blog/2023/05/12/removing-prototype-from-jenkins/) for details. Prototype will eventually be removed from Jenkins core. To prepare for this transition, this plugin must be migrated away from Prototype. The blog post contains instructions on how to find and eliminate usages of Prototype. In this plugin, the following usages of Prototype have been identified. This may not be an exhaustive list:

```
./src/main/resources/lib/categorizedview/catProjectView.jelly:99:    $$("table.categorizedSortable").each(function(e){
./src/main/resources/lib/categorizedview/catProjectViewRow.jelly:51:        document.observe("dom:loaded", function() {
./src/main/resources/org/jenkinsci/plugins/categorizedview/CategorizedJobsView/configure-entries.jelly:87:       var nestedElements = $$('SPAN.nested')
./src/main/resources/org/jenkinsci/plugins/categorizedview/CategorizedJobsView/configure-entries.jelly:89:         nestedElements.each(function(el) {
./src/main/resources/org/jenkinsci/plugins/categorizedview/CategorizedJobsView/configure-entries.jelly:90:           e.checked ? el.show() : el.hide();
./src/main/webapp/sectionToggle.js:3:    handle = $$("#handle_" + groupName).first();
./src/main/webapp/sectionToggle.js:12:    var handle = $$("#handle_" + group).first();
./src/main/webapp/sectionToggle.js:22:    $$('.' + group).each(
./src/main/webapp/sectionToggle.js:28:    var src = $$("#handle_" + group + " img").first().src;
./src/main/webapp/sectionToggle.js:30:    $$("#handle_" + group + " img").first().src = src;
./src/main/webapp/sectionToggle.js:35:    $$('.' + group).each(
./src/main/webapp/sectionToggle.js:38:            $(e).setOpacity(0)
./src/main/webapp/sectionToggle.js:47:    var src = $$("#handle_" + group + " img").first().src;
./src/main/webapp/sectionToggle.js:49:    $$("#handle_" + group + " img").first().src = src;
./src/main/webapp/sectionToggle.js:72:    localStorage.setItem("jenkins.categorized-view-collapse-state_" + viewName, Object.toJSON(groupStates));
./src/main/webapp/catSortable.js:57:        firstRow.each(function (cell) {
./src/main/webapp/catSortable.js:71:            firstRow.each(function (cell, i) {
./src/main/webapp/catSortable.js:98:                return $A(this.table.rows[0].cells);
./src/main/webapp/catSortable.js:197:                if (r.hasClassName("sorttop")) return 0;
./src/main/webapp/catSortable.js:198:                if (r.hasClassName("sortbottom")) return 2;
./src/main/webapp/catSortable.js:223:            rows.each(function (e) {
./src/main/webapp/catSortable.js:228:            this.arrows.each(function (e, i) {
```

CC @taksan

### PR DESCRIPTION
**Pull request by basil**
Fixes #31

Tested by setting breakpoints on every changed line and stepping through the code in the debugger to ensure the breakpoint was hit.

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/categorizedview/CategorizedJobsView/configure-entries.jelly [hunk 1]
```diff
@@ -84,10 +84,10 @@ THE SOFTWARE.
  <script>
    (function() {
      Behaviour.specify("#recurse", 'ListView', 0, function(e) {
-       var nestedElements = $$('SPAN.nested')
+       var nestedElements = document.querySelectorAll('SPAN.nested')
        e.onclick = function() {
-         nestedElements.each(function(el) {
-           e.checked ? el.show() : el.hide();
+         nestedElements.forEach(function(el) {
+           el.style.display = e.checked ? "" : "none";
```
Comment (basil) [inline src/main/resources/org/jenkinsci/plugins/categorizedview/CategorizedJobsView/configure-entries.jelly:90]: Copypasta of `core/src/main/resources/hudson/model/ListView/configure-entries-resources.js` from Jenkins core.

File: src/main/webapp/catSortable.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ var CategorizedSortable = (function () {
         if (!firstRow) return;
 
         // We have a first row: assume it's the header, and make its contents clickable links
-        firstRow.each(function (cell) {
+        firstRow.forEach(function (cell) {
```
Comment (basil) [inline src/main/webapp/catSortable.js:57]: All the changes in this file are loose copypasta from `war/src/main/webapp/scripts/sortable.js` in core.

File: src/main/webapp/catSortable.js [hunk 3]
```diff
@@ -358,7 +358,7 @@ var CategorizedSortable = (function () {
             var sortfn = this.caseInsensitive;
             if (itm.match(/^\d\d[\/-]\d\d[\/-]\d\d\d\d$/)) sortfn = this.date;
             if (itm.match(/^\d\d[\/-]\d\d[\/-]\d\d$/)) sortfn = this.date;
-            if (itm.match(/^[�$]/)) sortfn = this.currency;
+            if (itm.match(/^[£$]/)) sortfn = this.currency;
```
Comment (basil) [inline src/main/webapp/catSortable.js:361]: Not strictly related to Prototype.js, but side-porting jenkinsci/jenkins#7250 to this copypasta while I was here.

**Comment by basil** [comment]
CC @jenkinsci/core-security-review

**Review by MarkEWaite — APPROVED** [review]

**Review by yaroslavafenkin — COMMENTED** [review]
Looks OK security wise

**Pull request merged**